lnrpc/invoicesrpc/addinvoice.go

```diff
@@ -103,6 +103,11 @@ type AddInvoiceData struct {
 	// immediately upon receiving the payment.
 	HodlInvoice bool
 
+	// Amp signals whether or not to create an AMP invoice.
+	//
+	// NOTE: Preimage should always be set to nil when this value is true.
+	Amp bool
```

Comment:
So in practice, this works (creating an invoice even though we have the JIT invoice insertion path) because [we'll just ignore the error here](https://github.com/lightningnetwork/lnd/blob/8af00ab0cff83f09603657325e1d4bbd41b27a0e/invoices/invoiceregistry.go#L836) and continue?

Reply:
yep precisely!